src/OpenStudy/study_openinterface.jl

```diff
@@ -252,10 +252,12 @@ end
 
 function max_elements(data::Data, collection::String)
     raw = _raw(data)
+    
     if haskey(raw, collection)
         return length(raw[collection])
+    else
+        return 0
```

Comment:
undo

Reply:
I agree that returning `nothing` in this and other cases is a good practice. It is recommended by Julia's performance tips but also enhances the code's semantics.
In this case, returning `nothing` breaks things since many calls to `max_elements` do not worry about calling `isnothing` next.

I think this is the case for a major revision on missing/optional values as a whole since using zeroes, empty strings and 1990's january 1st to denote abscence are definitely bad karma purchases.

Having aditional boolean fields to indicate the presence of a value that could be `nothing` is also something we should avoid.

Reply:
You can open an issue so we can talk about these details later.
Zero is great for now indeed.